labs/src/main/scala/org/scalalabs/basic/lab03/PatternMatchingExercise.scala

```diff
@@ -47,7 +55,16 @@ object PatternMatchingExercise {
    *    - anything else, the function result is "Some Scala class"
    */
   def matchOnInputType(in: Any): String = {
-    error("fix me")
+    in match {
+      case string: String => "A string with length " + string.length
+      case number: Int => if (number > 0) "A positive integer" else "Some Scala class"
+      case personObject: Person => "A person with name: " + personObject.name
+      case sampleSequence: Seq[Any] => if (sampleSequence.length > 10) "Seq with more than 10 elements"
```

Comment:
I wrote something like this initially, but then I realized that it doesn’t work on infinite sequences (which are allowed; for example, `Stream continually 1` is an infinite sequence of 1s).